docs/development_guide.md

```diff
@@ -122,4 +140,29 @@ git commit --amend
 git push -f
 ```
 
-To ammend those changes to the original commit.
+To amend those changes to the original commit. Please note that using `--amend`
+creates a new commit and is a way of rewriting git history. In particular, if
+you have branched from the current branch to work on another change on top of
+the current one, `--amend` will make merging of these branches non-trivial.
+
+Another git approach that you can take is to create a new _review branch_ and use
+the `--squash` option of `git merge` to create one commit from all your changes
+and push that for review. For example, if you are on branch `foo` and ready to
+send a pull request, you can:
+```bash
+git checkout master
+git checkout -b foo_review
+git merge --squash foo
+git push origin foo_review
+```
+This approach is specially useful if you tend to do a lot of small commits
+during your feature development and like to keep them as checkpoints.
+
+### Continuous integration
```

Comment:
Do we want to make the docker images in the test project also public? I'm ok with that. It may even be a good idea for experimental images.
Otherwise, this is kind of confusing for an external user (since they don't have access to any of the images) and we should rephrase.

Reply:
I don't mind if we make those images public but even if we do that, can people who don't have access to the test project check the status of the builds? IIUC, making the images public is not enough to look at the build history dashboard. Or did you mean give broader public access to the test project itself?

I see your point about the potential confusion for an external contributor. I made the reference to the test project here for an internal contributor. I added a note to clarify the access issue.

Reply:
Thanks for rewording. I think this is good for now.
You are right about the build process not being public anyway.